### PR TITLE
Add dummy objects for new deployments to list in instance response.

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review. 
@smorgrav @oyving FYI.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java [hunk 1]
```diff
@@ -968,6 +970,17 @@ private void toSlime(Cursor object, Instance instance, DeploymentStatus status,
...
         }
+        // Add dummy values for not-yet-existent prod deployments.
+        status.jobSteps().keySet().stream()
+              .filter(job -> job.application().instance().equals(instance.name()))
+              .filter(job -> job.type().isProduction() && job.type().isDeployment())
+              .map(job -> job.type().zone(controller.system()))
+              .filter(zone -> ! instance.deployments().containsKey(zone))
+              .forEach(zone -> {
+                    Cursor deploymentObject = instancesArray.addObject();
+                    deploymentObject.setString("environment", zone.environment().value());
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Both console and dashboard expect that this object also contains `tenant`, `application` and `instance`.
Dashboard additionally expects that `url` is set.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Certainly won't set URL, as that would just point to a 404. So I can set the application ID, but this is surely pointless, as all three values are set on the grandparent object (the instance object container the array of deployments). I also intended for the missing `url` to serve as the indication that this is a dummy object.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Also, It seems to be the application id (tenant, application, instance) isn't set, unless it is a "recursive request"?

**Review by jonmv — COMMENTED** [review]
Urghj.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java (hunk 1 above)
Comment (freva) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Ah, yes, you're right. Let me fix the clients to make sure they can handle this change first (so they don't make the request that'll lead to 404).

**Review by freva — APPROVED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: 🙇

**Pull request merged**